src/ILLink.Shared/DataFlow/StructStack.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ILLink.Shared.DataFlow
+{
+	public struct StructStack<TValue> : IEquatable<StructStack<TValue>>
+		where TValue : IEquatable<TValue>
+	{
+		private Stack<TValue>? Stack;
+
+		private readonly int _count = 0;
+
+		public StructStack ()
+		{
+		}
+
+		public StructStack (StructStack<TValue> newStack)
+		{
+			Stack = newStack.Stack == null ? null : new Stack<TValue> (newStack.Stack);
+		}
+
+		public StructStack (int count)
```

Comment:
Is this meant as a "capacity" - as in how much it should allocate to begin with?
If so, the parameter should be called `capacity`. Depending on where we call this from, I might just allocate the stack immediately (since we're already asking for a set of slots).